source/dub/compilers/compiler.d

```diff
@@ -124,18 +126,63 @@ interface Compiler {
 		This method should be used by `Compiler` implementations to invoke the
 		compiler or linker binary.
 	*/
-	protected final void invokeTool(string[] args, void delegate(int, string) output_callback, string[string] env = null)
+	protected final void invokeTool(string[] args, void delegate(int, string) output_callback, string[string] env = null, Duration timeout = Duration.max)
 	{
 		import std.string;
 
+		string[] timeouted(string[] args) @safe pure
+		{
+			if (timeout)
+			{
+				import std.conv : to;
+				return ["timeout", timeout.total!"seconds".to!string] ~ args;
+			}
+			return args;
+		}
+
 		int status;
 		if (output_callback) {
-			auto result = executeShell(escapeShellCommand(args), env);
+			auto result = executeShell(escapeShellCommand(timeouted(args)), env); // TODO: avoid using timeoutArgs
```

Comment:
I don't think we should rely on timeout being installed, use D native timeout code like below here too, instead of relying on `timeout` as system command

Reply:
I explained above why that is not a good idea. The two are not synonymous. We should move in the other direction instead.